Tranga/MangaConnectors/HeanCms.cs

```diff
@@ -0,0 +1,421 @@
+
+        if (slug is null)
+        {
+            Log($"Got a null slug. {seriesData}");
+            return null;
+        }
+
+        Manga? manga = FetchMangaById(slug); // ! NB: FetchMangaById fetches the manga from the series endpoint
+
+        if (manga is null)
+        {
+            Log($"Failed to get manga from id. {slug}");
+            return null;
+        }
+
+        return manga;
+    }
+
+    private Manga ConvertSeriesToManga(JsonObject manga)
+    {
+        Log($"Converting series to manga");
+
+        string title = manga["title"]!.GetValue<string>();
+
+        List<string> authors = manga["author"]!.GetValue<string>().Split(" & ").ToList();
+
+        // description comes as html text, so we need to parse it
+        string descriptionHtml = manga["description"]!.GetValue<string>();
+        var htmlDoc = new HtmlDocument();
+        htmlDoc.LoadHtml(descriptionHtml);
+        string descriptionEscaped = htmlDoc.DocumentNode.InnerText;
+        string description = WebUtility.HtmlDecode(descriptionEscaped);
+
+        string altTitles = manga["alternative_names"]!.GetValue<string>();
+        Dictionary<string, string> altTitlesDict = new Dictionary<string, string>{{"0", altTitles}};
+
+        string[] tags = manga["tags"]!.AsArray().Where(tag => tag != null).Select(tag => tag!["name"]!.GetValue<string>()).ToArray();
+
+        string? coverUrl = manga["thumbnail"]!.GetValue<string>();
+        string? coverCacheName = null;
+        if (coverUrl is not null)
+            coverCacheName = SaveCoverImageToCache(coverUrl, RequestType.MangaCover);
+
+        Dictionary<string, string>? linksDict = null;
+
+        int? year = null;
+        string? yearStr = manga["release_year"]?.GetValue<string>();
+        if (yearStr != null && int.TryParse(yearStr, out int parsedYear))
+        {
+            year = parsedYear;
+        }
+
+        string? originalLanguage = null;    // TODO:
+
+        string status = manga["status"]!.GetValue<string>();
+
+        string publicationId = manga["series_slug"]!.GetValue<string>();
+
+        Log($"Got publicationId {publicationId}");
+
+        Manga.ReleaseStatusByte releaseStatus = Manga.ReleaseStatusByte.Unreleased;
+        switch (status.ToLower())
+        {
+            case "ongoing": releaseStatus = Manga.ReleaseStatusByte.Continuing; break;
+            case "completed": releaseStatus = Manga.ReleaseStatusByte.Completed; break;
+            case "hiatus": releaseStatus = Manga.ReleaseStatusByte.OnHiatus; break;
+            case "cancelled": releaseStatus = Manga.ReleaseStatusByte.Cancelled; break;
+        }
+
+        Manga pub = new(
+            title,
+            authors,
+            description,
+            altTitlesDict,
+            tags,
+            coverUrl,
+            coverCacheName,
+            linksDict,
+            year,
+            originalLanguage,
+            status,
+            publicationId,
+            releaseStatus
+        );
```

Comment:
Add field `websiteUrl` that links directly to the manga on the website.
Most of the sites drop the `api.` part.

Reply:
done